File: Assets/Script/Lib/Agent/Tool/VisionBinding.ts
// @preview-file off clear
import type { LLMConfig } from 'Agent/Utils';

/** Increment when a fixed vision route or its request profile changes. */
export const VISION_PROFILE_VERSION = 6;

export interface VisionBinding {
	provider: "deepseek" | "glm-coding-cn";
	model: string;
	url: string;
	apiKey: string;
	studioGateway?: boolean;
}

/** Only exact, reviewed service endpoints may reuse the current credential. */
export function resolveVisionBinding(config: LLMConfig): VisionBinding | undefined {
	if (config.apiKey.trim() === "") return undefined;
	if (config.studioGateway === true) {
		const vision = config.studioVision;
		if (!vision || config.apiKey !== "studio-agent") return undefined;
		const expectedModel = vision.provider === "deepseek" ? "deepseek-flash"
			: vision.provider === "glm-coding-cn" ? "glm-5.3-flash" : undefined;
		if (!expectedModel || vision.model !== expectedModel) return undefined;
		const [url] = string.gsub(vision.url.trim(), "/+$", "");
		if (!url.startsWith("https://") || !url.includes("/agent-host/") || !url.includes("/vision/")) return undefined;
		return { provider: vision.provider, model: vision.model, url, apiKey: config.apiKey, studioGateway: true };
	}
	const [url] = string.gsub(config.url.trim().toLowerCase(), "/+$", "");
	if (url === "https://api.deepseek.com/chat/completions" || url === "https://api.deepseek.com/v1/chat/completions") {
		return { provider: "deepseek", model: "deepseek-flash", url: "https://api.deepseek.com/v1/chat/completions", apiKey: config.apiKey };
	}
	if (url === "https://open.bigmodel.cn/api/coding/paas/v4/chat/completions") {
		return { provider: "glm-coding-cn", model: "glm-5.3-flash", url: "https://open.bigmodel.cn/api/paas/v4/chat/completions", apiKey: config.apiKey };
	}
	return undefined;
}

File: Assets/Script/Lib/Agent/Tool/VisionBinding.lua
-- [ts]: VisionBinding.ts
local ____lualib = require("lualib_bundle") -- 1
local __TS__StringTrim = ____lualib.__TS__StringTrim -- 1
local ____exports = {} -- 1
--- Increment when a fixed vision route or its request profile changes.
____exports.VISION_PROFILE_VERSION = 5 -- 5
--- Only exact, reviewed service endpoints may reuse the current credential.
function ____exports.resolveVisionBinding(config) -- 15
	if __TS__StringTrim(config.apiKey) == "" then -- 15
		return nil -- 16
	end -- 16
	local url = string.gsub( -- 17
		string.lower(__TS__StringTrim(config.url)), -- 17
		"/+$", -- 17
		"" -- 17
	) -- 17
	if url == "https://api.deepseek.com/chat/completions" or url == "https://api.deepseek.com/v1/chat/completions" then -- 17
		return {provider = "deepseek", model = "deepseek-v4-flash-vision-exp", url = "https://api.deepseek.com/v1/chat/completions", apiKey = config.apiKey} -- 19
	end -- 19
	if url == "https://open.bigmodel.cn/api/coding/paas/v4/chat/completions" then -- 19
		return {provider = "glm-coding-cn", model = "glm-5.3-flash", url = "https://open.bigmodel.cn/api/paas/v4/chat/completions", apiKey = config.apiKey} -- 22
	end -- 22
	return nil -- 24
end -- 15
return ____exports -- 15